examples/client/client.c

```diff
@@ -843,7 +844,9 @@ static int StartTLS_Init(SOCKET_T* sockfd)
     if (recv(*sockfd, tmpBuf, sizeof(tmpBuf)-1, 0) < 0)
         err_sys("failed to read STARTTLS command\n");
 
-    if (!XSTRCMP(tmpBuf, starttlsCmd[0])) {
+    XMEMCPY(buf, tmpBuf, XSTRLEN(starttlsCmd[0])+1);
```

Comment:
Before each `XMEMCPY()`, you need to check that `XSTRLEN(tmpBuf) >= XSTRLEN(starttlsCmd[the number])+1`, and check that `tmpBuf[XSTRLEN(starttlsCmd[the number]] == ' '` to confirm that the input has a space character where expected.

In the alternative you can skip the extra `buf` and the `XMEMCPY()`s, and use the simpler approach I showed in my earlier comment.